### DISCUSSION TITLE
How to disable SMS option in 2FA account access, without disabling Webauthn and 2FA entirely [obsolete]

### DISCUSSION THREAD

**Discussion by mk-fg** [community/community#48745] [Other Feature Feedback, Questions, & Ideas]
**EDIT**:
**THIS IS OBSOLETE** (as of April 2023), and no longer how that UI looks and works on github!
There is no "Reconfigure" button anymore, likely because it's now possible to do without it.

### Select Topic Area

Question

### Body

I have and want to have two-factor authentication enabled, but only with strong methods that are useful to me, like Webauthn, and disable terrible insecure methods that can easily be used to bypass these entirely.

But under "Password and Authentication" - "Two-factor authentication", github lists "SMS/Text message" as one allowed 2FA method, which is definitely not the one I want to have, as it is rather trivial and cheap to attach new SIM to any phone with local carriers (or mostly their minimally-paid staff rather).
I can click "Edit" on that SMS authentication method and enter a different phone, but I see no way to disable or remove it.

Is there a way to disable "SMS/Text message" second-factor authentication method?
And if not, why not? Maybe it can be a request for improvement in Github's currently-broken 2FA instead.

Here is how configuration screen looks for me (after clicking on "..." next to SMS 2FA method):
![221716114-e09dfe39-0789-4bdf-bedd-172b4335ce93](https://user-images.githubusercontent.com/227121/221784263-1ebf02e9-e62e-46c8-9c76-6fbbf372e9b3.jpg)

2FA is required by an org, but that doesn't mention SMS anywhere specifically, and 2FA shouldn't be assumed to be SMS.

Having SMS method force-enabled makes 2FA rather pointless for me - anyone, with at least some minor local knowledge and ~$50, can just ignore whatever I setup there, SMS-bypassing it all, and then do whatever they want on my github and wherever I have commit access to.

This is **not** about "Fallback SMS number" recovery, which I never setup and indeed [can be disabled](https://github.com/orgs/community/discussions/22948), but rather disabling (and ideally never even allowing) SMS as a 2FA method to log into my github account.

Thanks in advance for any assistance.

**Comment by mk-fg** [comment]
Was told a solution after asking this question elsewhere - apparently "Reconfigure" button in the top-right can replace mandatory SMS auth option with TOTP, which is still not what I'd use, but easier to forget and ignore, unlike leaving SMS one open.
It does not currently clear Webauthn, and can be used even if account is required to have 2FA by an org. Worked for me.

**Reply by hansbogert** [reply]
I'm probably blind, but where is that button?

**Reply by mk-fg** [reply]
Top-right of the "Two-factor authentication" box, to the right of its header, left of the green "Enabled" button there.
You can see it in the attached image above, at least where it was for me at the time of asking the question.

**Reply by mk-fg** [reply]
Just checked my settings and yeah, it's gone - looks like they did change how things work there.

**Reply by SparshSharmaOfficial** [reply]
@mk-fg 
What's the latest solution as I have same problem as the OP?

**Reply by mk-fg** [reply]
UI is different now, so I've no idea, unfortunately.

**Reply by samip5** [reply]
> @mk-fg What's the latest solution as I have same problem as the OP?

You need to add Authenicator App, before it will let you do pretty much anything despite if one has other forms of 2FA enabled.

**Reply by SparshSharmaOfficial** [reply]
@samip5 
Thanks but I already got it from someone: To disable SMS configuration, first set/configure using any Authenticator App and only then the option appears to remove SMS one. But I was trying to do the opposite. Unfortunately Github is poor at maths: law of commutative! 😂

**Comment by DNin01** [comment]
They also said [they were working on making it possible to use security keys as the primary second factor](https://github.blog/changelog/2022-11-21-updates-to-the-two-factor-authentication-setup-flow/).

> In the future, you'll also find security keys there as an option for initial setup on supported devices and browsers.

**Comment by kalbfled** [comment]
I have the same question/issue, and I also don't see a "Reconfigure" button.

**Reply by mk-fg** [reply]
Just checked - there isn't one for me either, guess it should be done in some different way.
Did edit the question/title to mark it as obsolete, as it's likely a global change on the site.

**Comment by kalbfled** [comment]
I added an authenticator app in addition to my security keys.  Once I did that, I had the option to disable SMS.

**Reply by DNin01** [reply]
That makes sense - security keys cannot be used as a "primary" authentication method yet, though GitHub said they are working on that feature.

**Comment by beeb** [comment]
I added security keys as 2FA and selected them as preferred method, but I still can't remove the SMS method. Only an "edit" button that allows to change the number, no way to disable it.

EDIT: after adding the Authenticator app again, I was able to disable the SMS option. But then I can't disable the Authenticator app option. Why can security keys not be the only selected option?

**Reply by pedrosimao** [reply]
Maybe they are afraid of people loosing their security keys? But anyway, that's why recovery codes exists right?

**Comment by VannTen** [comment]
I'm facing the same problem.
Passkeys are nice and all, but if I can't disable SMS 2FA, it's kinda pointless.

**Reply by pedrosimao** [reply]
Same here! Did you find a way?

**Comment by scheepers** [comment]
I swear if I have to solve that puzzle one more fkn time I'll go sterile.

I've added SSH keys, passkeys, my blood type and sacrificed the artworks of several small children.

Where is the SMS 2fa removal or 2fa reconfigure process, button or altar of mammon so I can access an organisation I belong to?

**Reply by PavlekJ** [reply]
it wasnt showing for me either when it was my only 2fa method, I thought I was going insane. 
Try configuring another 2fa method (ex. authenticator app), then you should see the three dots button in the SMS/Text message cell.

**Comment by lunvang** [comment]
Just found the solution!!
You have to add the "Authenticator app" two-factor method, and only then you'll be able to remove SMS.
I wish the UI would communicate that better, it was a pain to figure out.

**Reply by ianrtracey** [reply]
Wow this took me forever. Thank you!

**Comment by milandamen** [comment]
I want to use just WebAuthn and recover codes, and not have to set up TOTP as well.

Having to set up SMS or TOTP regardless of any other authentication method sucks. WebAuthn is more secure than other methods, and there's no reason to decrease security by requiring SMS/TOTP as well.

**Reply by samip5** [reply]
Agreed. That's absolutely annoying. Me with multiple 2FA options, including Github Mobile, Security Keys and Passkeys, only Authenication App allowed me to remove SMS which makes absolutely no sense.

**Comment by Paol0B** [comment]
Hi, try this
---


1. **Go to GitHub Security Settings**  
   👉 https://github.com/settings/security

2. **Scroll to "Two-Factor Authentication"**  
   You’ll see a list of your active 2FA methods (e.g., SMS, WebAuthn, TOTP).

3. **Find the SMS Method**  
   Click the **three dots (`⋯`)** next to the SMS method.

4. **Click “Remove” or “Edit”**  
   - If “Remove” is available, click it to delete SMS as a 2FA method.
   - If only “Edit” is available, you may need to **reconfigure your 2FA**.

5. **Use the “Reconfigure” Option (if needed)**  
   - Click **“Reconfigure”** at the top of the 2FA section.
   - Replace SMS with another method like TOTP (authenticator app) or WebAuthn.
   - This does **not disable 2FA** or remove WebAuthn.

6. **Confirm and Save**  
   Make sure you still have at least one active 2FA method (e.g., WebAuthn or TOTP).

---

##  Notes

- If your organization **requires 2FA**, GitHub will not let you remove all 2FA methods.
- You **must keep at least one method** active to avoid being locked out.

---

**Reply by samip5** [reply]
" Click the three dots (⋯) next to the SMS method." THat is NOT POSSIBLE even with multiple other factors (eg WebAuthn or/and Security Keys) unless Authenicator App is also used.

**Reply by caohy1988** [reply]
![image](https://github.com/user-attachments/assets/0ca4c708-0843-485e-bad0-17396e1f97d8)
I cannot find  three dots (⋯) next to the SMS method.  in my github account and cannot remove it , can you help me resolve it? thanks

**Comment by loboda-newfire** [comment]
What worked for me: 

- go to https://github.com/settings/security 
- scroll down to "Two-Factor Authentication"
- click on 3 dots at the end of "Two-Factor Authentication" title 
- click "Disable two-factor auth" 
- reconfigure Two-factor authentication from the scratch

**Comment by caohy1988** [comment]
![image](https://github.com/user-attachments/assets/0ca4c708-0843-485e-bad0-17396e1f97d8)
I cannot find  three dots (⋯) next to the SMS method.  in my github account and cannot remove it , can someone help me resolve it? thanks

**Reply by iennae** [reply]
Heya @caohy1988 , I ran into this today. What you need to do is add an Authenticator app and then the three dots option will appear. As soon as you remove the SMS/Text message option you'll then lose the three dots option. It appears that there is some kind of bug in this interface that requires one to have both Authenticator and SMS/Text message for the remove option to appear. Hope this helps!

**Comment by SnathanP** [comment]
Hey guys, this is kinda fucked up. I wanted to enforce 2FA and "secure 2FA" on my organization (eg, no SMS) and its causing trouble for newcomers: if they have 2FA enabled with SMS, basically, we can do nothing. Not cool :/

**Reply by iennae** [reply]
heya @SnathanP the path to fixing this is have them add an Authenticator app and then the three dots option will appear. Then they can remove the SMS option which will then mean they lose the three dots option. I ran into this today and I hope this helps.

**Comment by wherka-ama** [comment]
Guys, I've just helped one lost soul to sort this issue out. I'm not sure if you've tried that before, but a simple way to deal with it is to ensure you have a Authenticator and/or Mobile flow active and then start the sign in in **Incognito mode**. 

That guarantees you don't have any session details, cookies etc. It seems to be pretty effective in exposing the Edit/Delete on the SMS/Text message.

**Comment by shinybrightstar** [comment]
Hi @mk-fg   👋,

Currently, GitHub requires you to first set up either an authenticator app (TOTP) or SMS as your primary 2FA method. Once you have done that, you can add GitHub Mobile or Security Keys as an additional 2FA option, which allows you to verify sign-ins directly from your device.

After enabling 2FA, GitHub provides a set of [recovery codes](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/recovering-your-account-if-you-lose-your-2fa-credentials). Download these codes and keep them in a secure, offline location. **These are extremely important**—if you lose access to all your 2FA methods and your recovery codes, it may be difficult to restore access to your account.

You can find more information in the [official GitHub documentation](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication).

Thank you!

**Comment by james-ecd** [comment]
I had to set up an Authentication app as a 2fa option before it gave me the option to disable sms. Once I had setup an authentication app, three dots appeared on the the sms option giving an option to disable

**Comment by wherka-ama** [comment]
@james-ecd : indeed. You do need to have these two methods in place in order to kick the 3rd one out. This is what we've noted as well.

Summarising:
- in order to remove the **SMS/Text** or in fact any other auth method, you need to have two other methods configured on top of it(**GitHub Mobile + Authenticator app**)
- in some cases we have some issues with weird state of a **three dots button** - it is shown, but not usable; in such case you may try using **incognito mode** of your browser

**Comment by MahdiBM** [comment]
This implementation by GitHub is absolutely bonkers.
It's clear they're not taking security seriously and just trying to look like they do.

How do any of these make any sense?

* I have set up (multiple) security keys, but no no no you have to set up "Authenticator app" to be able to remove SMS auth?! Worse yet, this is not pointed out anywhere?
* Now that I've deleted SMS auth, there is no option to delete "Authenticator app" when I already have security keys set up?!
* Now that I've set up both auth-app and security keys (both use the security keys anyway), I can't remove my GitHub mobile app as an auth method?!

All the way through the multi-auth set up, GitHub is like "No you don't need to remove your weakest link! Please don't!" while trying to obstruct the way to remove those weaker links.

I even contacted GitHub support a year+ ago. Got no response back of course. Until I read this thread about how to remove SMS auth.

**Reply by milandamen** [reply]
Yep they don't care and don't understand security and UX.

**Comment by ChaseKnowlden** [comment]
Just add your security key and your authenticator app to your GitHub account.

**Reply by MahdiBM** [reply]
I don't see the relation of this comment to my comment to be honest. I thought I was clear I already have both those set up. I thought I was also clear that I want to delete the weaker links now as I have no need for them ("GitHub mobile" and "authenticator app"). So really struggling to see how this comment is supposed to be helpful. Perhaps it wasn't directed towards my comment?

**Reply by tbates** [reply]
@ChaseKnowlden didn't add what would probably make his suggestion make sense to you, which is "then and only then will delete become available for SMS"

**Comment by yhxd123** [comment]
in 2026, you have to first download a auth app (e.g. Google/Microsfot Auth) from app store then register that app as auth method, then you are allowed to delete text message option. It kinda sucks but make sense since Github enforces auth app for a long time (since 2016)

**Reply by tbates** [reply]
But sub-optimal that there is no note on the SMS pane indicating that delete (required for some organizations) is unavailable until AFTER you activate a 2AFC authenticator (even if you have other methods enabled).
PS: on Mac, passwords.app can store 2AFC challenges: right click on the QR code to set up, (or copy+paste the key code)

**Reply by tbates** [reply]
sub note: set up on Mac doesn't work for Safari: have to use chrome for this one step, as POST is broken on GitHub/safari
